presto-main/src/main/java/com/facebook/presto/sql/rewrite/MaterializedViewOptimizationRewrite.java

```diff
@@ -87,4 +97,34 @@ protected Node visitQuery(Query query, Void context)
             return query;
         }
     }
+
+    private static Query optimizeQueryUsingMaterializedView(
+            Metadata metadata,
+            Session session,
+            SqlParser sqlParser,
+            AccessControl accessControl,
+            Query node)
+    {
+        MaterializedViewCandidateExtractor materializedViewCandidateExtractor = new MaterializedViewCandidateExtractor(session, metadata);
+        materializedViewCandidateExtractor.process(node);
+        Map<QualifiedObjectName, List<QualifiedObjectName>> baseTableToMaterializedViewNameMap = materializedViewCandidateExtractor.getMaterializedViewCandidatesForTable();
+        // TODO: Select the most compatible and efficient materialized view for query rewrite optimization https://github.com/prestodb/presto/issues/16431
+        // TODO: Refactor query optimization code https://github.com/prestodb/presto/issues/16759
+
+        Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViewDefinitionMap = new HashMap<>();
+        baseTableToMaterializedViewNameMap.forEach((baseTable, materializedViewNames) -> {
+            for (QualifiedObjectName materializedViewName : materializedViewNames) {
+                MaterializedViewStatus materializedViewStatus = metadata.getMaterializedViewStatus(session, materializedViewName);
+                if (materializedViewStatus.isNotMaterialized() && isMaterializedViewDataConsistencyEnabled(session)) {
+                    session.getRuntimeStats().addMetricValue(MANY_PARTITIONS_MISSING_IN_MATERIALIZED_VIEW_COUNT, NONE, 1);
```

Comment:
It's worth nothing this stat will represent a different number with this change- previously we would run the `isNotMaterialized()` check *after* going through the entire rewrite, now we will do the check before. This means `addMetricValue()` will be triggered more frequently.

Reply:
This doesn't look useful here, can we move it to part where we know we have rewritten a query specification?

Reply:
As per our offline discussion I'm adding a todo and resolving this for now.